geoportal/geoportailv3_geoportal/lib/lingua_extractor.py

```diff
@@ -0,0 +1,378 @@
+                    config, dbsession_name, **dbsession_config
+                )
+
+            try:
+                self._extract_messages(db_session)
+
+            except ProgrammingError as e:
+                print(
+                    colorize(
+                        "ERROR! The database is probably not up to date "
+                        "(should be ignored when happen during the upgrade)",
+                        RED,
+                    )
+                )
+                print(colorize(e, RED))
+                if os.environ.get("IGNORE_I18N_ERRORS", "FALSE") != "TRUE":
+                    raise
+
+        except NoSuchTableError as e:
+            print(
+                colorize(
+                    "ERROR! The schema didn't seem to exists "
+                    "(should be ignored when happen during the deploy)",
+                    RED,
+                )
+            )
+            print(colorize(e, RED))
+            if os.environ.get("IGNORE_I18N_ERRORS", "FALSE") != "TRUE":
+                raise
+        except OperationalError as e:
+            print(
+                colorize(
+                    "ERROR! The database didn't seem to exists "
+                    "(should be ignored when happen during the deploy)",
+                    RED,
+                )
+            )
+            print(colorize(e, RED))
+            if os.environ.get("IGNORE_I18N_ERRORS", "FALSE") != "TRUE":
+                raise
+
+        return self.messages
+
+        # return [Message(None, 'msgid', None, [], u'', u'', (filename, 1))]
+
+    def _insert_attribute(self, attribute, location):
+        if attribute not in self.fields:
+            print(f'Adding attribute {attribute}')
+            self.fields.add(attribute)
+            self.messages.append(
+                Message(
+                    None,
+                    u'%(name)s' % {'name': attribute},
+                    None,
+                    [],
+                    "",
+                    "",
+                    location,
+                )
+            )
+
+    # abstract prototype
+    def _extract_messages(self, db_session):
+        raise(Exception('not implemented'))
+
+
+class LuxembourgESRILegendExtractor(LuxembourgExtractor):  # pragma: no cover
+    """
+    ESRI legend extractor
+    """
+    extensions = [".ini"]
+
+    def _extract_messages(self, db_session):
+        print('Entering into ESRI extractor')
+
+        from c2cgeoportal_commons.models import DBSession  # pylint: disable=import-outside-toplevel
+        from geoportailv3_geoportal.models import LuxLayerInternalWMS
+
+        results = (DBSession.query(LuxLayerInternalWMS)
+                   .filter(LuxLayerInternalWMS.rest_url.isnot(sqlalchemy.null)))
+        print("%d ESRI layers to parse" % results.count())
+
+        for result in results:
+            self._load_result(result)
```

Comment:
`dbsession` is a already a parameter of the fonction, why importing it ?

Also we are using a recent version of python, you can drop the old style print and do `print(f"{results.count()} ESRI layers to parse")` But I'm maybe to strict :sweat_smile:

Reply:
yes